resources/install.rb

```diff
@@ -18,17 +18,20 @@
 #
 
 include IISCookbook::Helper
+include IISCookbook::WindowsFeatureHelper
 
 property :source, String
 property :additional_components, Array, default: []
+property :install_method, Symbol, required: false, equal_to: [:windows_feature_dism, :windows_feature_powershell, :windows_feature_servermanagercmd], default: :windows_feature_dism
```

Comment:
Rather than having the multiple `to_sym` calls in various places, change this property type to `[String, Symbol]` and add a coerce of `proc { |p| p.to_sym}` to the property definition so you'll always have a symbol further on down the call stack. It also opens this up to greater flexibility where passing a symbol isn't as easy as it should be (instantiating from roles where symbol doesn't exist in json for one).

Reply:
yep

Reply:
I can certainly do this for the install resource. However 99% of the `tosym` calls are for calling into the windows_feature resource directly from a recipe, so I don't have to call it there, since we are taking the value directly from the attribute.
if we expect the attribute to be a symbol, it won't work in test kitchen and we can't test it properly